packages/core-data/src/components/RelatedList.js

```diff
@@ -107,7 +112,17 @@ const RelatedList = (props: Props) => {
             key={item.id}
             className='flex items-center'
           >
-            { props.renderItem(item) }
+            { props.onClick
```

Comment:
I think this was the point of making the `renderItem` prop pretty generic. It should allow the consumer to decide how to render not only the item content, but also any wrappers:

```
<RelatedList
  renderItem={(item) => (
    <button
      className='flex items-center py-1 px-2'
      onClick={() => action('click')}
      type='button'
    >
      { item }
    </button>
  )}
/>
```

```
<RelatedList
  renderItem={(item) => (
    <a
      className='flex items-center py-1 px-2'
      href={`/items/${item.id}`}
    >
      { item.name }
    </a>
  )}
/>
```

Reply:
Hm, should I refactor this so the conditional `button` rendering happens when the `renderItem` callback is passed via `RelatedOrganizations`, `RelatedPeople`, etc.? Or should I not render the button at all, and just apply the `onClick` callback directly to the `li` (which the linter will complain about, with good reason because there's no out-of-the-box indication to the user that the `li` component is interactive)?

Alternatively, maybe I could refocus this PR to add a `renderItem` prop to all of these components instead of `onClick`? That would also solve Sapientia's needs would also allow Sapientia to use a more semantic link component instead of a button.

Reply:
> Alternatively, maybe I could refocus this PR to add a renderItem prop to all of these components instead of onClick? That would also solve Sapientia's needs would also allow Sapientia to use a more semantic link component instead of a button.

I think this is probably the way to go. It would provide a more flexibility, although it would require duplicating some of the code in the consumer in order to add a `button` or `a` wrapper. Probably worth the trade off thought.